mantidproject/mantid
# Indirect Interfaces need to work with Project Recovery after a crash

**Description of work.**
The indirect inelastic GUIs can be unstable. To help minimize the problems this causes we want the GUI's to work with project recovery.
To do this the history of the output workspaces need to be fixed by:
- Making sure outputs are stored in the ADS (before being deleted at an appropriate time)
- Setting the correct algorithms to be children

**To test:**
One possible test:
1. Interfaces Menu->Indirect->Data Analysis
2. Go on I(Q, t) tab
3. Load Data
4. Click 'Run' and wait
5. Wait for Project Recovery to save and then crash mantid
6. History should have been recovered!

Fixes  #23144

Does this update require release notes?


---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
